src/main/groovy/org/shipkit/gradle/plugin/PluginDiscoveryTask.java

```diff
@@ -1,8 +1,13 @@
-package org.shipkit.internal.gradle.plugin;
+package org.shipkit.gradle.plugin;
 
+import com.gradle.publish.PluginBundleExtension;
 import org.gradle.api.DefaultTask;
 import org.gradle.api.tasks.TaskAction;
+import org.shipkit.internal.gradle.plugin.tasks.PluginDiscovery;
 
+/**
+ * Discovers gradle plugins and adds them to the {@link PluginBundleExtension}.
```

Comment:
Can you write a little bit more how it discovers the plugins? I think it would be really useful for our users to understand how exactly it happens.